src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5429,9 +5438,26 @@ public override BoundNode VisitDiscardExpression(BoundDiscardExpression node)
 
         public override BoundNode VisitThrowExpression(BoundThrowExpression node)
         {
-            var result = base.VisitThrowExpression(node);
-            SetUnknownResultNullability();
-            return result;
+            VisitThrow(node.Expression);
+            return null;
+        }
+
+        public override BoundNode VisitThrowStatement(BoundThrowStatement node)
+        {
+            VisitThrow(node.ExpressionOpt);
+            return null;
+        }
+
+        private void VisitThrow(BoundExpression expr)
+        {
+            VisitRvalue(expr);
+            // We treat `throw null` like an explicit `throw new System.NullReferenceException()`. The user is being explicit enough, so a warning would be annoying.
+            if (!expr.IsLiteralNull())
+            {
+                var exceptionType = TypeSymbolWithAnnotations.Create(compilation.GetWellKnownType(WellKnownType.System_Exception), NullableAnnotation.NotNullable);
```

Comment:
>var exceptionType = TypeSymbolWithAnnotations.Create(compilation.GetWellKnownType(WellKnownType.System_Exception), NullableAnnotation.NotNullable); [](start = 16, length = 147)

Why do this complicated analysis with fake conversions? We could simply check if the value can be null and complain about that.

Reply:
I agree. That makes more sense. Thanks